conn_id_generator.go

```diff
@@ -76,6 +76,9 @@ func (m *connIDGenerator) Retire(seq uint64) error {
 	if !ok {
 		return nil
 	}
+	if connID.Equal(sentWithDestConnID) {
+		return qerr.NewError(qerr.FrameEncodingError, fmt.Sprintf("tried to retire connection ID %d (%s), which was used as the Destination Connection ID on this packet", seq, connID))
```

Comment:
This should be a PROTOCOL_VIOLATION, see https://github.com/quicwg/base-drafts/issues/3860.